File: Selection_Sort/Python/madisonmay/selection_sort.py
def selection_sort(l):
    length = len(l)
    for i in range(length):
        min_index = i
        for j in range(i, length):
            if l[j] < l[min_index]:
                min_index = j
        if min_index != i:
            l[i], l[min_index] = l[min_index], l[i]
    return l

if __name__ == '__main__':
    # test cases
    l = list(range(10, 0, -1))
    print('Unsorted list: ' + str(l));
    l = selection_sort(l)
    print('Sorted list: ' + str(l))

File: Selection_Sort/Go/jcla1/selection_sort.go
package selectionSort

func SelectionSort(arr []int) []int {
	length := len(arr)
	var minIndex int

	for i := range arr {
		minIndex = i
		for j := i; j < length; j++ {
			if arr[j] < arr[minIndex] {
				minIndex = j
			}
		}
		if minIndex != i {
			arr[i], arr[minIndex] = arr[minIndex], arr[i]
		}
	}
	return arr
}
